services/app-web/src/hooks/useContractForm.ts

```diff
@@ -0,0 +1,192 @@
+import { useState, useEffect} from 'react'
+import { usePage } from "../contexts/PageContext"
+import {
+    CreateContractInput,
+    useFetchContractQuery,
+    useCreateContractMutation,
+    useUpdateContractDraftRevisionMutation,
+    ContractDraftRevisionFormDataInput,
+    UpdateInformation,
+    Contract
+} from '../gen/gqlClient'
+import { recordJSException } from '../otelHelpers'
+import { handleApolloError } from '../gqlHelpers/apolloErrors'
+import { ApolloError } from '@apollo/client'
+import type { InterimState } from '../pages/StateSubmission/ErrorOrLoadingPage'
+
+
+type  UseContractForm = {
+    draftSubmission?: Contract
+    unlockInfo?: UpdateInformation
+    showPageErrorMessage: string | boolean
+    previousDocuments?: string[]
+    updateDraft: (
+        input: Contract
+    ) => Promise<Contract | Error>
+    createDraft: (input: CreateContractInput) => Promise<Contract | Error>
+    interimState?:  InterimState
+    submissionName?: string
+}
+
+const useContractForm = (contractID?: string): UseContractForm => {
+    // Set up defaults for the return value for hook
+    let interimState: UseContractForm['interimState'] = undefined // enum to determine what Interim UI should override form page
+    let previousDocuments: UseContractForm['previousDocuments'] = [] // used for document upload tables
+    let draftSubmission: UseContractForm['draftSubmission'] = undefined // form data from current package revision, used to load form
+    let unlockInfo: UseContractForm['unlockInfo'] = undefined
+    const [showPageErrorMessage, setShowPageErrorMessage] = useState<boolean | string>(false) // string is a custom error message, defaults to generic of true
+    const { updateHeading } = usePage()
+    const [pkgNameForHeading, setPkgNameForHeading] = useState<string | undefined>(undefined)
+
+    useEffect(() => {
+        updateHeading({ customHeading: pkgNameForHeading })
+    }, [pkgNameForHeading, updateHeading])
+
+    const { 
+        data: fetchResultData,
+        error: fetchResultError,
+        loading: fetchResultLoading
+     } = useFetchContractQuery({
+        variables: {
+            input: {
+                contractID: contractID ?? 'new-draft'
+            }
+        },
+        skip: !contractID
+    })
+    const [createFormData] = useCreateContractMutation()
+
+    const createDraft: UseContractForm['createDraft']  = async (
+        input: CreateContractInput
+    ): Promise<Contract | Error> => {
+        setShowPageErrorMessage(false)
+        const {populationCovered,programIDs,riskBasedContract, submissionType, submissionDescription, contractType} = input
+        if(populationCovered === undefined || contractType === undefined) {
+            return new Error('wrong')
```

Comment:
I think we could fix this now - I believe this should be handled with form validations